src/Payments.sol

```diff
@@ -805,18 +805,73 @@ contract Payments is
         }
     }
 
-    function processOneTimePayment(
+    function calculateAndPayFees(
+        uint256 amount,
+        address token,
+        address operator,
+        uint256 commissionRateBps
+    ) internal returns (uint256 netPayeeAmount, uint256 paymentFee, uint256 operatorCommission) {
+        // Calculate payment contract fee (if any) based on full amount
+        paymentFee = 0;
+        if (PAYMENT_FEE_BPS > 0) {
+            paymentFee = (amount * PAYMENT_FEE_BPS) / COMMISSION_MAX_BPS;
+        }
+        
+        // Calculate amount remaining after contract fee
+        uint256 amountAfterPaymentFee = amount - paymentFee;
+        
+        // Calculate operator commission (if any) based on remaining amount
+        operatorCommission = 0;
+        if (commissionRateBps > 0) {
+            operatorCommission = (amountAfterPaymentFee * commissionRateBps) / COMMISSION_MAX_BPS;
+        }
+        
+        // Calculate net amount for payee
+        netPayeeAmount = amountAfterPaymentFee - operatorCommission;
+        
+        // Credit operator (if commission exists)
+        if (operatorCommission > 0) {
+            Account storage operatorAccount = accounts[token][operator];
+            operatorAccount.funds += operatorCommission;
+        }
+        
+        // Track platform fee
+        if (paymentFee > 0) {
+            // Check if this is the first fee for this token
+            if (accumulatedFees[token] == 0) {
+                feeTokens.push(token);
+            }
```

Comment:
Ditto the other PR, this will break if we withdraw the funds down to zero.

Reply:
Fixed this bug in the other PR. Will rebase.